lib/express_validator.js

```diff
@@ -1,262 +1,335 @@
-        }
-        req._validationErrors.push(error);
-
-        if (req.onErrorCallback) {
-          req.onErrorCallback(msg);
-        }
-        return this;
+    locations.forEach(function(location) {
+      req['check' + capitalCase(location)] = function(param, failMsg) {
+        return new ValidatorChain(param, failMsg, req, location, options);
       };
+    });
+
+    req.checkFiles = function(param, failMsg) {
+      return new ValidatorChain(param, failMsg, req, 'files', options);
+    };
 
-      var methods = [];
+    req.checkHeaders = function(param, failMsg) {
+      if(param === 'referrer') {param = 'referer';}
+      return new ValidatorChain(param, failMsg, req, 'headers', options);
+    };
 
-      Object.keys(validator).forEach(function(methodName) {
-        if (!methodName.match(/^to/) && sanitizers.indexOf(methodName) === -1) {
-          methods[methodName] = function() {
-            var args = [value].concat(Array.prototype.slice.call(arguments));
-            var isCorrect = validator[methodName].apply(validator, args);
+    req.check = function(param, failMsg) {
+      return new ValidatorChain(param, failMsg, req, locate(req, param), options);
+    };
 
-            if (!isCorrect) {
-              errorHandler(failMsg || 'Invalid value');
-            }
 
-            return methods;
-          };
-        }
-      });
+    req.filter = req.sanitize;
+    req.assert = req.check;
+    req.validate = req.check;
+
+    next();
+  };
+};
 
-      Object.keys(_options.customValidators).forEach(function(customName) {
-        methods[customName] = function() {
-          var args = [value].concat(Array.prototype.slice.call(arguments));
-          var isCorrect = _options.customValidators[customName].apply(null, args);
+/**
+ * validate an object using a schema
+ *
+ * @method validateParams
+ * @param  {Object}       schema  schema of validations
+ * @param  {Object}       obj     object to validate
+ * @param  {Object}       options options containing custom validators & errorFormatter
+ * @return {object[]}             array of errors
+ */
 
-          if (!isCorrect) {
-            errorHandler(failMsg || 'Invalid value');
-          }
 
-          return methods;
-        };
-      });
+/**
+ * Validates and handles errors, return instance of itself to allow for chaining
+ *
+ * @method makeValidator
+ * @param  {string}          methodName
+ * @param  {object}          container
+ * @return {function}
+ */
 
-      methods['notEmpty'] = function() {
-        return methods.isLength(1);
-      };
+function makeValidator(methodName, container) {
+  return function() {
+    var args = [];
+    args.push(this.value);
+    args = args.concat(Array.prototype.slice.call(arguments));
 
-      methods['len'] = function() {
-        return methods.isLength.apply(methods.isLength, Array.prototype.slice.call(arguments));
-      };
+    var isValid =  (this.skipValidating) ? true : container[methodName].apply(container, args);
```

Comment:
There are two space in front of this equals sign. Also, it might be more readable to assign `container[methodName].apply(container, args)` to a variable and then just have `var isValid =  this.skipValidating || newVariable;`